objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+        card.removeButton(getButtonIndex(card,"g3Minus"))
+        impendTable[tbl.guid].energy[3] = 0
+        updateEnergyDisplay(tbl.guid)
+    end
+end
+
+function updateG3Buttons(guid)
+    local card = getObjectFromGUID(guid)
+    if card then
+        local plus_color = impendTable[guid].energy[3] == 1 and {1,1,0,1} or {1,1,1,1}
+        local plus_ind = getButtonIndex(card,"g3Plus")
+        card.editButton({
+            index = plus_ind,
+            color = plus_color,
+        })
+        local minus_color = impendTable[guid].energy[3] == -1 and {1,1,0,1} or {1,1,1,1}
+        local minus_ind = getButtonIndex(card,"g3Minus")
+        card.editButton({
+            index = minus_ind,
+            color = minus_color,
+        })
+    end
+end
+
+function g3CardsSelected()
+    local count = 0
+    for _,tbl in pairs(impendTable) do
+        if tbl.energy[3] ~= 0 then
+            count = count + 1
+        end
+    end
+    return count
+end
+
+function g3Plus(card)
+    local selected = g3CardsSelected()
+    local guid = card.guid
+    if impendTable[guid].energy[3] == 1 then
+        impendTable[guid].energy[3] = 0
+    elseif impendTable[guid].energy[3] == -1 or selected < 2 then
+        impendTable[guid].energy[3] = 1
+    end
+    updateEnergyDisplay(guid)
+    updateG3Buttons(guid)
+    updateSave()
+end
+
+function g3Minus(card)
+    local selected = g3CardsSelected()
+    local guid = card.guid
+    if impendTable[guid].energy[3] == -1 then
+        impendTable[guid].energy[3] = 0
+    elseif impendTable[guid].energy[3] == 1 or selected < 2 then
+        impendTable[guid].energy[3] = -1
+    end
+    updateEnergyDisplay(guid)
+    updateG3Buttons(guid)
+    updateSave()
+end
+
+function createAllCardButtons()
+    createAllG3Buttons()
+    for guid,_ in pairs(impendTable) do
+        local card = getObjectFromGUID(guid)
+        if card then
+            Wait.time(function()
+                createCardButtons(card)
+                updateEnergyDisplay(guid)
+                if g3Selected and impendTable[guid].turnSelected ~= Global.getVar("turn") then
+                    createG3Buttons(card)
+                    updateG3Buttons(guid)
+                end
+            end, 0.01)
+        end
+    end
+    updateSave()
+end
+
+------------------
+-- other functions
+------------------
+
+function getButtonIndex(obj,str) -- needed for cards that already have buttons so can be edited normally
+    if obj.getButtons() then
+        for _,tbl in pairs(obj.getButtons()) do
+            if tbl.click_function == str then
+                return tbl.index
+            end
+        end
+    end
+    return nil
+end
+
+function timePasses()
+    local dances = Global.call("getSpirit", {name = spiritName})
+    for _,tbl in pairs(impendTable) do
+        local var = tbl.energy[1] + tbl.energy[2] + tbl.energy[3]
+        if var >= tbl.maxEnergy and tbl.turnSelected ~= Global.getVar("turn") - 1 then
```

Comment:
what the turn comparison part of this conditional supposed to be handling?

Reply:
If a card is impended, with enough energy to meet its threshold it doesn't go into play until the following turn since the cards are mechanically added after energy gain, which is when they go into play. This if statement is for wiping impend data for any cards that are no longer impended at the end of the turn, which are the cards that had enough energy to surpass their threshold, but weren't put into play this turn. The turn comparison is just to take out any cases where the energy requirement was hit turn one, else they'd be removed from impend before they are ever played.

Reply:
> which are the cards that had enough energy to surpass their threshold, but weren't put into play this turn

when would this happen, just the player forgetting to play the card?

> the energy requirement was hit turn one

this is for the case where say the player impends a 0 cost power card, but you still need to wait till the following turn to actually play it?